polyfactory/factories/base.py

```diff
@@ -541,16 +541,18 @@ def get_constrained_field_value(cls, annotation: Any, field_meta: FieldMeta) ->
                     pattern=constraints.get("pattern"),
                 )
 
-            with suppress(ValueError):
+            try:
                 collection_type = get_collection_type(annotation)
+            except ValueError:
+                collection_type = None
+            if collection_type is not None:
```

Comment:
Not important, but we can avoid creating `collection_type = None` manually and use `try`/`except`/`else` instead:

```py
            try:
                collection_type = get_collection_type(annotation)
            except ValueError:
                pass
            else:
                if collection_type == dict:
```
This may prevent further mistakes if someone use `collection_type = None`  unintentionally somewhere in the further code.

Reply:
Yeah, this is definitely a possibility, but I just don't like empty `except ValueError` here. That was the reason I inititally used a `with supress(ValueError)`.